Catalog page main filter

AlinaPalyvoda opened the pull request.

AlinaPalyvoda committed:
feat:
- create CatalogMainFilter (intermediate stage of work)

AlinaPalyvoda committed:
Merge branch 'develop' into feature/catalog-page-main-filter

AlinaPalyvoda committed:
Merge branch 'develop' into feature/catalog-page-main-filter

AlinaPalyvoda committed:
feat: added components:
- FilterAccordion,
- CatalogFilterPrice,
- CatalogFilterTourists,
- CatalogFilterRegion,
- CatalogMainFilter

AlinaPalyvoda committed:
Merge branch 'develop' into feature/catalog-page-main-filter

AlinaPalyvoda committed:
feat: work on the adaptability of components

AlinaPalyvoda committed:
moving filter title from page to CatalogMainFilter

AlinaPalyvoda committed:
Merge branch 'develop' into feature/catalog-page-main-filter

AlinaPalyvoda committed:
fixed bug of toggling all checkboxes

AlinaPalyvoda committed:
feat: added ShowButton

AlinaPalyvoda committed:
feat: added reset button

AlinaPalyvoda committed:
refactor: CatalogFilterPrice

AlinaPalyvoda committed:
feat: added snapshot tests for:
- FilterAccordion,
- CatalogFilterPrice,
- CatalogFilterTourists,
- CatalogFilterRegion,
- CatalogMainFilter

Ahmed-Fahmy-Dev commented on client/src/components/CatalogMainFilter/CatalogMainFilter.js:
```diff
@@ -0,0 +1,82 @@
+
+const CatalogMainFilter = () => (
+  <FilterWrapper>
+    <Typography variant="h3">Filters</Typography>
```
```suggestion
    <Typography variant="h3">Filter</Typography>
```

AlinaPalyvoda replied:
done

Ahmed-Fahmy-Dev commented on client/src/components/CatalogMainFilter/CatalogMainFilter.js:
```diff
@@ -0,0 +1,82 @@
+      </Grid>
+    </Grid>
+
+    <ShowButton>Show</ShowButton>
```
```suggestion
    <ShowButton>Filter</ShowButton>
```

AlinaPalyvoda replied:
done

Ahmed-Fahmy-Dev commented on client/src/components/CatalogMainFilter/CatalogMainFilter.test.js:
```diff
@@ -0,0 +1,15 @@
+    const { asFragment } = render(<CatalogMainFilter />);
+    expect(asFragment()).toMatchSnapshot();
+  });
+});
```
this is a unit test. please also mock the below components:

CatalogFilterPrice
CatalogFilterTourists
CatalogFilterRegion


please contact me for more clarification

Ahmed-Fahmy-Dev commented on client/src/components/CatalogFilterPrice/CatalogFilterPrice.js:
```diff
@@ -0,0 +1,56 @@
+import React from 'react';
```
```suggestion
import React, { useState } from 'react';
```

AlinaPalyvoda replied:
done

Ahmed-Fahmy-Dev commented on client/src/components/CatalogFilterPrice/CatalogFilterPrice.js:
```diff
@@ -0,0 +1,56 @@
+}));
+
+const CatalogFilterPrice = () => {
+  const [value, setValue] = React.useState([200, 600]);
```
```suggestion
  const [value, setValue] = useState([200, 600]);
```

AlinaPalyvoda replied:
done

Ahmed-Fahmy-Dev commented on client/src/components/CatalogFilterPrice/CatalogFilterPrice.test.js:
```diff
@@ -0,0 +1,14 @@
+import { render } from '@testing-library/react';
+import CatalogFilterPrice from './CatalogFilterPrice';
+
+jest.mock('@mui/material/Box', () => ({ children }) => <div>{children}</div>);
+jest.mock('@mui/material/Slider', () => ({ children }) => <span>{children}</span>);
+jest.mock('@mui/material/InputBase', () => ({ children }) => <input>{children}</input>);
+
+
+describe('CatalogFilterPrice snapshot test', () => {
+  test('should CatalogFilterPrice match snapshot', () => {
+    const { asFragment } = render(<CatalogFilterPrice />);
+    expect(asFragment()).toMatchSnapshot();
+  });
+});
```
please mock FilterAccordion component as a div

Ahmed-Fahmy-Dev commented on client/src/components/CatalogFilterTourists/CatalogFilterTourists.js:
```diff
@@ -0,0 +1,81 @@
+  },
+}));
+
+const NumberButton = ({ value, children }) => {
```
please move this component and all its logic into a separate file.

AlinaPalyvoda replied:
done

Ahmed-Fahmy-Dev commented on client/src/components/CatalogFilterTourists/CatalogFilterTourists.test.js:
```diff
@@ -0,0 +1,12 @@
+import { render } from '@testing-library/react';
+import CatalogFilterTourists from './CatalogFilterTourists';
+
+jest.mock('@mui/material/ToggleButton', () => ({ children }) => <button>{children}</button>);
+jest.mock('@mui/material/Unstable_Grid2', () => ({ children }) => <div>{children}</div>);
+
+describe('CatalogFilterTourists snapshot test', () => {
+  test('should CatalogFilterTourists match snapshot', () => {
+    const { asFragment } = render(<CatalogFilterTourists />);
+    expect(asFragment()).toMatchSnapshot();
+  });
+});
```
please mock FilterAccordion and NumberButton

Ahmed-Fahmy-Dev commented on client/src/components/CatalogFilterRegion/CatalogFilterRegion.js:
```diff
@@ -0,0 +1,55 @@
+import React from 'react';
```
```suggestion
import React, { useState } from 'react';
```

AlinaPalyvoda replied:
done

Ahmed-Fahmy-Dev commented on client/src/components/CatalogFilterRegion/CatalogFilterRegion.js:
```diff
@@ -0,0 +1,55 @@
+const regions = ['region1', 'region2', 'region3', 'region4'];
+
+const CatalogFilterRegion = () => {
+  const [checked, setChecked] = React.useState(Array(regions.length).fill(false));
```
```suggestion
  const [checked, setChecked] = useState(Array(regions.length).fill(false));
```

AlinaPalyvoda replied:
done

Ahmed-Fahmy-Dev commented on client/src/components/CatalogFilterRegion/CatalogFilterRegion.js:
```diff
@@ -0,0 +1,55 @@
+    });
+  };
+
+  const children = (
+    <Box sx={{ display: 'flex', flexDirection: 'column', ml: 3 }}>
+      {regions.map((el) => (
+        <FormControlLabel
+          key={el}
+          label={el}
+          control={<Checkbox id={el} checked={checked[regions.indexOf(el)]} onChange={handleChange} />}
+        />
+      ))}
+    </Box>
+  );
```
not a very preferred method to declare children in such way.  very complicated and non readable.

AlinaPalyvoda replied:
This method is offered by material ui.
![image](https://user-images.githubusercontent.com/107770819/201555913-28be6137-5c2a-4594-8643-a2ed209f4f3b.png)

What exactly is hard to read?

Ahmed-Fahmy-Dev commented on client/src/components/CatalogFilterRegion/CatalogFilterRegion.test.js:
```diff
@@ -0,0 +1,13 @@
+import { render } from '@testing-library/react';
+import CatalogFilterRegion from './CatalogFilterRegion';
+
+jest.mock('@mui/material/Box', () => ({ children }) => <div>{children}</div>);
+jest.mock('@mui/material/Checkbox', () => ({ children }) => <span>{children}</span>);
+jest.mock('@mui/material/FormControlLabel', () => ({ children }) => <label>{children}</label>);
+
+describe('CatalogFilterRegion snapshot test', () => {
+  test('should CatalogFilterRegion match snapshot', () => {
+    const { asFragment } = render(<CatalogFilterRegion />);
+    expect(asFragment()).toMatchSnapshot();
+  });
+});
```
please mock FilterAccordion and all functions also using jest.fn

Ahmed-Fahmy-Dev reviewed (commented):
Great Job !

HIMDanny commented on client/src/components/CatalogFilterTourists/CatalogFilterTourists.js:
```diff
@@ -0,0 +1,81 @@
+      value={value}
+      selected={selected}
+      onChange={() => {
+        setSelected(!selected);
```
what about using previous value instead of reverting `selected`?
```suggestion
        setSelected(prev => !prev);
```

AlinaPalyvoda replied:
I used the provided method from material ui. But your proposal makes sense. Done

HIMDanny commented on client/src/components/CatalogMainFilter/CatalogMainFilter.js:
```diff
@@ -0,0 +1,82 @@
+import CloseIcon from '@mui/icons-material/Close';
+import { CatalogFilterPrice, CatalogFilterTourists, CatalogFilterRegion } from '../index';
+
+const FilterWrapper = styled(Box)(({ theme }) => ({
+  display: 'flex',
+  flexDirection: 'column',
```
on a note:
You can get rid of these two properties just using `Stack` instead of `Box`
```suggestion
const FilterWrapper = styled(Stack)(({ theme }) => ({

```

AlinaPalyvoda replied:
Done

HIMDanny commented on client/src/components/FilterAccordion/FilterAccordion.js:
```diff
@@ -0,0 +1,58 @@
+  padding: '0 0 40px',
+});
+
+const AccordionTitle = styled((props) => <Typography variant="body1" {...props} />)(() => ({
```
`Typography` has "body1" variant by default
```suggestion
const AccordionTitle = styled(Typography)(() => ({
```

AlinaPalyvoda replied:
Done

HIMDanny commented on client/src/pages/CataloguePage/CataloguePage.js:
```diff
@@ -1,19 +1,42 @@
+          </Stack>
+        </Grid>
+
+        <FilterContainer>
+          <CatalogMainFilter />
+        </FilterContainer>
```
what about moving filter on top of cards? We will have a greater amount of them so it would not be handy to use filter

HIMDanny reviewed (commented):
I also agree with @Ahmed-Fahmy-Dev comments

In total, good job!

AlinaPalyvoda committed:
fix:
- fixed filter components,
- moved the OutlinedToggleButton from CatalogFilterTourists  to a separate file

AlinaPalyvoda committed:
Merge branch 'develop' into feature/catalog-page-main-filter

AlinaPalyvoda committed:
- update CatalogMainFilter snapshot test,
- added OutlinedToggleButton snapshot test

Ahmed-Fahmy-Dev committed:
hotfix:   snapshot test fix
- mocked components to achieve clean unit test

Ahmed-Fahmy-Dev committed:
hotfix: snapshot test fix
- mocked components to achieve clean unit test

Ahmed-Fahmy-Dev committed:
hotfix: snapshot test fix
- mocked components to achieve clean unit test

Ahmed-Fahmy-Dev reviewed (approved):
fixed snapshot tests by myself.

approved
